include/bluetooth/audio/cap.h

```diff
@@ -86,6 +137,210 @@ struct bt_cap_cb {
  */
 int bt_cap_find_ccid_service_service(struct bt_conn *conn, uint8_t ccid);
 
+/**
+ * @brief Discovers audio support on a remote device.
+ *
+ * This will discover the Common Audio Service (CAS) on the remote device, to
+ * if the remote device supports the Common Audio Profile.
+ *
+ * @param conn Connection to a remote server.
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_cap_unicast_discover(struct bt_conn *conn);
+
+/** Type of CAP set */
+enum bt_cap_set_type {
+	/** The set is an ad-hoc set */
+	BT_CAP_SET_TYPE_AD_HOC,
+	/** The set is a CSIP Coordinated Set */
+	BT_CAP_SET_TYPE_CSIP,
+};
+
+struct bt_cap_unicast_audio_start_param {
+	/** The type of the set */
+	enum bt_cap_set_type type;
+
+	/** The number of set members in @p members and number of streams in @p streams */
+	uint8_t count;
+
+	union {
+		/**
+		 * @brief List of connection pointers if @p type is BT_CAP_SET_TYPE_AD_HOC
+		 *
+		 * The number of pointers in this array shall be at least @p count
+		 */
+		struct bt_conn **members;
```

Comment:
I'd rather want to avoid using `group` as a `group` in ISO (and also the Zephyr audio API) means something different from a set. 

If we want to expose an API to create and populate these groups/sets, then we should probably implement them as linked lists instead of arrays, but that can be done. 

Do you see any advantages of exposing an API to create these groups/sets instead of having it as a application allocated struct?

Reply:
linked list is fine. 

I think it would make things easier, if whenever user have to talk to stack in the context of group, it would be easier to have a defined group representation.
Otherwise, once user needs to construct array of members, in the `bt_audio_unicast_group` it has list of streams. That's only my feeling.

Reply:
We could use list as a parameter instead of an array of pointers. This is a bit different from existing BT APIs that takes more than 1 pointer as input, but certainly not something I would be against. 

The downside of supporting the `slist` implementation, is that we will need to append `sys_snode_t node` to the structs that we want to put into a list. For something like `bt_conn` that is opaque, we will need to create a wrapper around it, but we could implement something like 

```c
struct bt_device_list_item {
    union {
        struct bt_conn *conn;
        struct bt_csis_client_set_member *member
    }

    sys_snode_t node;
}
```

So using a list approach will require a new struct, and then the application can use the `slist` API to create the list, such as 

```c

struct bt_device_list_item list_items[cnt] = { /* initialize array */ ); /* or just create these structs on the fly */

sys_slist_t list;

for (int i = 0; i < cnt; i++) {
    sys_slist_append(&list, &list_items[i].node);
}

bt_cap_unicast_audio_start_param(&list);
```

as opposed to the current solution:
```c
struct bt_conn conns[cnt] = { /* initialize array */ };

bt_cap_unicast_audio_start_param(&conns);
```